src/services/item/plugins/app/appAction/repository.ts

```diff
@@ -48,6 +48,13 @@ export const AppActionRepository = AppDataSource.getRepository(AppAction).extend
       where: { item: { id: In(itemIds) }, member: { id: memberId } },
       relations: { item: true },
     });
+    // todo: should use something like:
+    // but this does not work. Maybe related to the placement of the IN ?
+    // const appActions = await this.createQueryBuilder('actions')
+    //   .innerJoinAndSelect('actions.item', 'item', 'item.id IN (:...itemIds)', { itemIds })
+    //   .where('actions.member = :memberId', { memberId })
+    //   .andWhere('actions.item IN (:...itemIds)', { itemIds })
+    //   .getMany();
```

Comment:
I think it's missing a join for the member? Also from the entity definition, both member and item can actually be null.
Which means that we can't actually differentiate between an action that is not linked to an Item and an action linked to a soft deleted item...

Maybe in this case we can use `withDeleted: true` in the find options? Maybe we still want the item info when we look at the actions?

But if not it's a tricky. We don't want a inner join because we will lose the actions which are not linked to a member or a item.

Reply:
Good catch about the missing join on member ! But I am not sure it is allowed for the item or member to be null ? `nullable: false` means it can't be null right ? And we have a cascade on delete, so my guess would be that we may not want to retrieve appactions that belong to trashed items ... maybe @pyphilia can help.